sonar-db/src/main/java/org/sonar/db/component/SnapshotDao.java

```diff
@@ -68,17 +66,16 @@ public SnapshotDto selectOrFailById(DbSession session, long id) {
     return executeLargeInputs(snapshotIds, mapper(dbSession)::selectByIds);
   }
 
-  @CheckForNull
-  public SnapshotDto selectLastSnapshotByComponentUuid(DbSession session, String componentUuid) {
-    return mapper(session).selectLastSnapshot(componentUuid);
+  public Optional<SnapshotDto> selectLastSnapshotByComponentUuid(DbSession session, String componentUuid) {
+    return Optional.ofNullable(mapper(session).selectLastSnapshotByComponentUuid(componentUuid));
   }
 
-  public List<SnapshotDto> selectLastSnapshotByComponentUuids(DbSession dbSession, List<String> componentUuids) {
-    return componentUuids.isEmpty() ? emptyList() : mapper(dbSession).selectLastSnapshotByComponentUuids(componentUuids);
+  public Optional<SnapshotDto> selectLastSnapshotByRootComponentUuid(DbSession session, String componentUuid) {
+    return Optional.ofNullable(mapper(session).selectLastSnapshotByRootComponentUuid(componentUuid));
   }
 
-  public boolean hasLastSnapshotByComponentUuid(DbSession session, String componentUUid) {
-    return mapper(session).countLastSnapshotByComponentUuid(componentUUid) > 0;
+  public List<SnapshotDto> selectLastSnapshotsByRootComponentUuids(DbSession dbSession, List<String> componentUuids) {
```

Comment:
why is this method not using `DatabaseUtils.executeLargeInputs` ?
In which case, parameter `componentUuids` would have type `Collection` rather than `List` (see `executeLargeInputs`)